src/main/java/com/google/idea/gn/GnFoldingBuilder.kt

```diff
@@ -0,0 +1,50 @@
+package com.google.idea.gn
+
+import com.google.idea.gn.psi.impl.GnBlockImpl
+import com.google.idea.gn.psi.impl.GnCollectionImpl
+import com.intellij.lang.ASTNode
+import com.intellij.lang.folding.FoldingBuilderEx
+import com.intellij.lang.folding.FoldingDescriptor
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import java.util.*
+
+class GnFoldingBuilder : FoldingBuilderEx(), DumbAware {
+    override fun buildFoldRegions(root: PsiElement, document: Document, quick: Boolean): Array<FoldingDescriptor> {
+        val descriptors: MutableList<FoldingDescriptor> = ArrayList()
+
+        //Block
+        val blocks = PsiTreeUtil.findChildrenOfType<GnBlockImpl>(root, GnBlockImpl::class.java)
```

Comment:
nit: Explicit type arguments can be omitted.

Reply:
Explicit type arguments have been removed. 👌